jax/lax/lax.py

```diff
@@ -1020,25 +1020,40 @@ def transpose(operand: Array, permutation: Sequence[int]) -> Array:
   else:
     return transpose_p.bind(operand, permutation=permutation)
 
-def reduce(operand: Array, init_value: Array, computation: Callable,
-           dimensions: Sequence[int]) -> Array:
+def reduce(operand: Union[Array, Tuple[Array]], init_value: Union[Array, Tuple[Array]],
```

Comment:
Should we be polymorphic between list and tuple here, as in e.g. `Union[Array, Tuple[Array], List[Array]]`? I don't have a strong preference, but I've found users often expect tuple/list polymorphism.

Reply:
Good point... what do you think about `Union[Array, Sequence[Array]]` with an `isinstance(operand, typing.Sequence)` check? That returns False for numpy & jax arrays, and True for Tuples and Lists.

Reply:
SGTM! I didn't know that kind of `isinstance` check worked.